rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateFilter.spec.ts

```diff
@@ -0,0 +1,179 @@
+import { mount, VueWrapper } from "@vue/test-utils";
+import DateFilter from "../dateFilter.vue";
+import DateTimePicker from "../dateTimePicker.vue";
+import _ from "lodash";
+let idCounter = 0;
+jest.spyOn(_, "uniqueId").mockImplementation(() => `uniqueId_${idCounter++}`);
+
+interface MountOptions {
+  props?: Record<string, unknown>;
+  slots?: { default?: string };
+  global?: {
+    stubs?: Record<string, unknown>;
+    mocks?: Record<string, unknown>;
+  };
+}
+const mountDateFilter = (options: MountOptions = {}): VueWrapper<any> => {
+  return mount(DateFilter, {
+    props: {
+      modelValue: { enabled: false, datetime: "" },
+      ...options.props,
+    },
+    slots: {
+      default: options.slots?.default || "slotContent",
+    },
+    global: {
+      stubs: {
+        btn: true,
+        DateTimePicker: true,
+        Dropdown: {
+          template: '<div class="dropdown"><slot name="dropdown"></slot></div>',
+        },
+      },
+      mocks: {
+        $t: (msg) => msg,
+      },
+    },
+  });
+};
+describe("DateFilter", () => {
+  beforeEach(() => {
+    idCounter = 0;
+    jest.spyOn(_, "uniqueId").mockImplementation(() => {
+      const id = `uniqueId_${idCounter}`;
+      idCounter++;
+      return id;
+    });
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  describe("Initialization", () => {
+    it("initializes with correct data", () => {
+      const wrapper = mountDateFilter();
+      expect(wrapper.vm.uid).toBeTruthy();
+      expect(wrapper.vm.enabled).toBe(false);
+      expect(wrapper.vm.datetime).toBe("");
+    });
+    it("sets uid correctly", () => {
+      const wrapper = mountDateFilter();
+      expect(wrapper.vm.uid).toBe(`uniqueId_${idCounter - 1}`);
+    });
```

Comment:
better to check that the input has the right id (which is the uid) instead of checking wrapper.vm.uid